src/app/page.tsx

```diff
@@ -1,101 +1,426 @@
-import Image from "next/image";
+"use client";
+import {
+  Box,
+  Button,
+  Grid2,
+  Stack,
+  ThemeProvider,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import { useState } from "react";
+import { CloudUpload } from "@mui/icons-material";
+import FileElement from "@/components/FileElement";
+import FileUploaded from "@/classe/File";
+
+/**
+ * Represents the state of a dropzone component.
+ *
+ * @interface DropzoneState
+ * @property {boolean} visible - Indicates whether the dropzone is visible.
+ * @property {string | null} image_url - The URL of the image in the dropzone, or null if no image is present.
+ * @property {number} [fillPercentage] - Optional property representing the fill percentage of the dropzone.
+ */
+interface DropzoneState {
+  visible: boolean;
+  image_url: string | null;
+  fillPercentage?: number;
+}
+
+/**
+ * Interface representing an image load event in a React application.
+ * Extends the React.SyntheticEvent interface with a target of type HTMLImageElement.
+ *
+ * @interface ImageLoadEvent
+ * @extends {React.SyntheticEvent<HTMLImageElement>}
+ * @property {HTMLImageElement} target - The target element of the event, which is an HTMLImageElement.
+ */
+interface ImageLoadEvent extends React.SyntheticEvent<HTMLImageElement> {
+    target: HTMLImageElement;
+}
+
+/**
+ * Interface representing the dimensions of a parent element.
+ *
+ * @property {number} width - The width of the parent element.
+ * @property {number} height - The height of the parent element.
+ */
+interface ParentDimensions {
+    width: number;
+    height: number;
+}
+
+function Home() {
+  const theme = useTheme();
+  const [dropzoneState, setDropzoneState] = useState<DropzoneState>({
+    visible: false,
+    image_url: null,
+    fillPercentage: 0,
+  });
+
+  const [uploadedFiles, setUploadedFiles] = useState<FileUploaded[]>([]);
+
+/**
+ * Updates the state of the dropzone component.
+ *
+ * @param {boolean} show - Determines whether the dropzone should be visible.
+ * @param {string} image_url - The URL of the image to be displayed in the dropzone.
+ */
```

Comment:
Please, avoid doctrings for such subfunctions.

Reply:
I know you are right but one of the first point for the refactor is to make documentation for every function in the code so that new developer could easily enter in the code. Should I make a separate .md file to explain each function?

Reply:
I would say such extensive commenting requirements are only for low level university courses. They're more for TAs to have an easier time understanding homework. Short comments about why the function exists is more important. For what the function actually does it should be obvious from the code implementation. The input requirements are also already handled by TypeScript hints.